packages/x-date-pickers/src/internals/utils/date-utils.ts

```diff
@@ -110,7 +110,11 @@ export const applyDefaultDate = (
   return value;
 };
 
-export const areDatesEqual = (utils: MuiPickersAdapter, a: PickerValidDate, b: PickerValidDate) => {
+export const areDatesEqual = (
+  utils: MuiPickersAdapter,
+  a: PickerValidDate | null,
+  b: PickerValidDate | null,
```

Comment:
Another nitpick: Would this work?
```suggestion
  a: PickerValue,
  b: PickerValue,
```

Reply:
Done :+1:

Reply:
Are you sure: https://github.com/mui/mui-x/pull/15434/commits/e5406e7172dcd1fe1c9aaf12b9bcd524b25247f7? 🙈